src/app/server/Mdns.cpp

```diff
@@ -119,31 +119,19 @@ CHIP_ERROR AdvertiseOperational()
     return mdnsAdvertiser.Advertise(advertiseParameters);
 }
 
-/// Set MDNS commissioner advertisement
-CHIP_ERROR AdvertiseCommisioner()
+/// Overloaded utility method for commissioner and commissionable advertisement
+// This method is used for both commissioner discovery and commissionable node discovery since
+// they share many fields.
+CHIP_ERROR Advertise(bool commissionableNode, CommissioningMode mode)
 {
-    return Advertise(false);
-}
+    bool commissioningMode = (mode == CommissioningMode::kEnabled || mode == CommissioningMode::kEnabledAsAdditionalCommissioning);
+    bool additionalCommissioning = (mode == CommissioningMode::kEnabledAsAdditionalCommissioning);
```

Comment:
As a followup: we should make these APIs clearer so someone does not "fix" additionalCommissioning to be based on whether we already have an opcert.  That's not what it means.

Reply:
> Marking as 'change requested' to fix up the (bool,bool) arguments - I feel strongly that we allowed that to go in and need to fix it. Too much bitrot that makes reviewing/maintainance harder.
> 
> Should be a small change.

ok, please take a look at the latest thanks

Reply:
I've added many comments to the use of these fields. I think it would be useful to have a global state somewhere which tracks commissioning mode for the device (and whether its basic or enhanced). That would allow us to remove these arguments completely.

Reply:
Isn't `RendezvousServer` the entity that should push this state to Mdns?  It by definition knows whether the node is commissionable or not.  I don't think the Mdns code is well-suited to pull this state from some central location instead.  Much of the 'what-state-are-we-in' logic in the configuration manager is broken anyway.

I think the problem is more that we've embedded all sorts of inference logic in the Mdns code.  Why is Advertise called from `StartServer` instead of `RendezvousServer`?  Why does the Mdns code need to infer what should be advertised on behalf of another module in the sdk?

It would be nice if the Mdns code provided public methods like these, but didn't actually call them: `AdvertiseCommissionableNode`, `AdvertiseCommissionableNodeAP`, `SuspendCommissionableNodeAdvertisement`.

If these were instead called from `RendezvousServer`, I think the Mdns code would be much simpler.